src/main/java/org/opentripplanner/model/calendar/CalendarServiceData.java

```diff
@@ -72,6 +73,24 @@ public void add(CalendarServiceData other) {
     }
   }
 
+  public Optional<LocalDate> getFirstDate() {
+    return serviceDatesByServiceId
+      .values()
+      .stream()
+      .filter(list -> !list.isEmpty())
+      .map(list -> list.get(0))
+      .min(LocalDate::compareTo);
```

Comment:
This methods do not need to return optional either; An empty calendar should never be processed - so you may throw an exception instead of returning the optional.

Reply:
Fixed in https://github.com/opentripplanner/OpenTripPlanner/pull/5162/commits/71727bd94f50c1cd51818e460ce21cff77965f2c, I wasn't sure which implementation was faster but it shouldn't matter as both are fast enough and your suggestion in simpler.